### DISCUSSION TITLE
The relation between task-canal/task-canal-adapter and task-syncer

### DISCUSSION THREAD

**Discussion by uicosp** [cubefs/compass#155] [Q&A] [answered]
After read the [docs](https://github.com/cubefs/compass/blob/main/document/manual/deployment_zh.md), i know that the task-canal/task-canal-adapter modules are used to sync schedule data to compass.
but, when i inspect the task-syncer module, there is a MessageConsumer to listen the canal kafka topic to sync data from scheduler to compass also.

So, my question is what's the difference between task-canal-adapter and task-syncer? why there are two diff sync logic in two modules?

**Comment by zebozhuang** [comment, marked as answer]
> After read the [docs](https://github.com/cubefs/compass/blob/main/document/manual/deployment_zh.md), i know that the task-canal/task-canal-adapter modules are used to sync schedule data to compass. but, when i inspect the task-syncer module, there is a MessageConsumer to listen the canal kafka topic to sync data from scheduler to compass also.
> 
> So, my question is what's the difference between task-canal-adapter and task-syncer? why there are two diff sync logic in two modules?

(Question 1)Task-canal-adapter synchronizes the raw data to compass locally while task-syncer synchronize the bin log data as event to trigger work flow. And one important point there is initialization in task-syncer module which will synchize data too but it will convert to compass table data but not origin scheduler raw table data. 

(Question 2) Actually it is best to combine two modules into one and we have not optimize it yet since it needs more time.

**Reply by uicosp** [reply]
i see, thanks for your reply
